sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_custom_form_client_async.py

```diff
@@ -0,0 +1,330 @@
+from .._generated.models import TrainRequest, TrainSourceFilter
+from .._policies import CognitiveServicesCredentialPolicy
+from .._response_handlers import (
+    prepare_unlabeled_result,
+    prepare_labeled_result,
+)
+from .._generated.models import AnalyzeOperationResult, Model
+from .._helpers import get_pipeline_response, get_content_type, POLLING_INTERVAL
+from .._models import (
+    ModelInfo,
+    ModelsSummary,
+    CustomModel,
+    CustomLabeledModel,
+)
+if TYPE_CHECKING:
+    from .._credential import FormRecognizerApiKeyCredential
+    from .._models import (
+        ExtractedPage,
+        ExtractedLabeledForm
+    )
+
+
+class CustomFormClient(object):
+    """CustomFormClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of FormRecognizerApiKeyCredential if using an API key.
+    :type credential: ~azure.ai.formrecognizer.FormRecognizerApiKeyCredential
+    """
+
+    def __init__(
+            self,
+            endpoint: str,
+            credential: "FormRecognizerApiKeyCredential",
+            **kwargs: Any
+    ) -> None:
+        self._client = FormRecognizer(
+            endpoint=endpoint,
+            credential=credential,
+            authentication_policy=CognitiveServicesCredentialPolicy(credential),  # TODO: replace with core policy
+            **kwargs
+        )
+
+    @distributed_trace_async
+    async def begin_training(
+            self,
+            source: str,
+            source_prefix_filter: Optional[str] = "",
+            include_sub_folders: Optional[bool] = False,
+            **kwargs: Any
+    ) -> CustomModel:
+        """Create and train a custom model. The request must include a source parameter that is an
+        externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri).
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff'. Other type of content is ignored.
+
+        :param str source: An Azure Storage blob container URI.
+        :param str source_prefix_filter: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob Uri, use the prefix to restrict sub
+            folders for training.
+        :param bool include_sub_folders: A flag to indicate if sub folders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed.
+        :return: CustomModel
+        :rtype: ~azure.ai.formrecognizer.CustomModel
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        content_type = kwargs.pop("content_type", "application/json")
```

Comment:
Suggestion (nitpick): we can just do a `kwargs.setdefault('content_type', 'application/json') and drop `content_type=content_type` parameter when creating the `TrainRequest`.

Reply:
Can't do that without popping it before I create the poller, else we get an unexpected kwarg. In fact, I'll remove setting content-type here completely because the generated code defaults it to "application/json".

Reply:
Ok - sounds good!